src/ui/StatusEditor.svelte

```diff
@@ -29,29 +29,24 @@
     }
 
     const _onStatusChange = () => {
-        // Use statusSymbol to find the status to save to editableTask.status
-        const selectedStatus: Status | undefined = statusOptions.find((s) => s.symbol === statusSymbol);
-        if (selectedStatus) {
-            editableTask.status = selectedStatus;
-        } else {
-            console.log(`Error in EditTask: cannot find status with symbol ${statusSymbol}`);
-            return;
-        }
+        const newStatus = new Status(JSON.parse(selectedStatus).configuration);
```

Comment:
I feel that the use of `JSON.parse()` here is a bit of a complication.

It was not previously a requirement that names of statuses are safe to save in JSON - and now it is.

What happens if there are any characters in the status name that are problematical with JSON saving and parsing?

Also, previously, editing now creates new instances of statuses in the status registry - whereas previously it was a simple object look-up.

Reply:
I suppose a less negative response would be to say that this PR would benefit from a bunch of new tests using custom statuses with a bunch of corner cases to try and break the JSON round-tripping - but I really don’t think that would be a good use of your time or mine. 😄

Reply:
> It was not previously a requirement that names of statuses are safe to save in JSON - and now it is.
> 
> What happens if there are any characters in the status name that are problematical with JSON saving and parsing?

True, it was previously a requirement that the statuses are saved as status symbols though. Do you think these 2 are equivalent complications?

> Also editing now creates new instances of statuses in the status registry - whereas previously it was a simple object look-up.

I'm sorry but I fail to see how...I wrote this code to test this:

```ts
    const _onStatusChange = () => {
        console.log(StatusRegistry.getInstance().registeredStatuses);
        const newStatus = new Status(JSON.parse(selectedStatus).configuration);
        console.log(StatusRegistry.getInstance().registeredStatuses);
        ...
    }
```

And the length of `registeredStatuses` was the same. I'm probably missing something here.

> I suppose a less negative response would be to say that this PR would benefit from a bunch of new tests using custom statuses with a bunch of corner cases to try and break the JSON round-tripping - but I really don’t think that would be a good use of your time or mine. 😄

Agreed. As per my understanding of [JSON](https://ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf), it can parse and stringily both ways any parameters.

---

Generally speaking, if you consider this or any other PR detrimental to Tasks in any way, feel free to close it. I throw ideas, but at the end of the day, I trust your perspective more than mine =)

Reply:
> True, it was previously a requirement that the statuses are saved as status symbols though. Do you think these 2 are equivalent complications?

I think my comments have been unclear, and I am unsure I can explain it effectively by typing... Let's talk about it next time we pair, if you feel it's worthwhile.

> Generally speaking, if you consider this or any other PR detrimental to Tasks in any way, feel free to close it. I throw ideas, but at the end of the day, I trust your perspective more than mine =)

That is a very gracious and generous statement - thank you! ❤️ 

I am going to close this one, I'm afraid.